packages/jest-runtime/src/transform.js

```diff
@@ -293,6 +294,30 @@ const transformSource = (
   return result;
 };
 
+const generateEmptyCoverage = (
+  filename: Path,
+  config: Config,
+  content: string
+): Object => {
+  const transform = getTransformer(filename, config);
+  let source = content;
+
+  if (shouldTransform(filename, config)) {
+    if (transform.canInstrument && typeof transform.getEmptyCoverage === 'function') {
+      return transform.getEmptyCoverage(content, filename, config, {
```

Comment:
Yes, exactly. Not sure how important visual accuracy is for uncovered code, but it can look pretty busted without this, so you'd probably get a couple issues now and then. Also prevents the branch/etc count from jumping up/down depending on whether the file executed or not.